packages/@aws-cdk/pipelines/lib/helpers-internal/graph.ts

```diff
@@ -382,4 +382,22 @@ function projectDependencies<A>(dependencies: Map<GraphNode<A>, Set<GraphNode<A>
 
 export function isGraph<A>(x: GraphNode<A>): x is Graph<A> {
   return x instanceof Graph;
-}
\ No newline at end of file
+}
+
+function deduplicateDependencies<A>(dependencies: GraphNode<A>[]) {
+  dependencies.sort((a, b) => a.uniqueId.localeCompare(b.uniqueId));
+  return dedupeGraphNodes(dependencies);
+}
+
```

Comment:
While this isn't wrong, it feels overly complex in this case.

Why don't we use the built-in `Set` ?

```ts
function deduplicateDependencies<A>(dependencies: GraphNode<A>[]) {
  const ret = new Array<GraphNode<A>>();
  const seen = new Set<GraphNode<A>>();

  for (const dep of dependencies) {
    if (seen.has(dep)) { continue; }
    seen.add(dep);
    ret.push(dep);
  }

  return ret;
}
```

Or if you want to be fancy, with a generator helper function:

```ts
function deduplicateDependencies<A>(dependencies: GraphNode<A>[]) {
  const seen = new Set<GraphNode<A>>();
  return Array.from(generate());
 
  function* generate() {
    for (const dep of dependencies) {
      if (seen.has(dep)) { continue; }
      seen.add(dep);
      yield dep;
      ret.push(dep);
    }
  }
}
```